.github/workflows/gh-pages-main.yml

```diff
@@ -0,0 +1,34 @@
+name: App build CI/CD to main branch
+
+on:
+  push:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          persist-credentials: false
```

Comment:
q: why do we need this?

Reply:
this is to push changes to the `gh-pages` branch as it counts as a `git commit` to `gh-pages` branch